t/t2801-top-cmd.t

```diff
@@ -12,6 +12,25 @@ testssh="${SHARNESS_TEST_SRCDIR}/scripts/tssh"
 
 export FLUX_URI_RESOLVE_LOCAL=t
 
+# To ensure no raciness in tests below, ensure the job-list
+# module knows about submitted jobs in desired states
+JOB_WAIT_ITERS=100
+job_list_wait_state() {
+	expected=$1
```

Comment:
I'm not sure I see how `job_list_wait_state()` is supposed to work. It looks like it takes one argument (the expected state?) and assumes there is only one active job, but below it seems to be called with two arguments (`jobid` and `state`). Is there just a missing argument (perhaps to `flux jobs`) here?

Also, I'm seeing 
```
./t2801-top-cmd.t: line 23: [: 0: unary operator expected
```
when running the current version of this test.

Also, a more broad question - the eventual consistency issue seems to come up frequently with job-list. It would be nice if there was a way to ensure job-list state is synchronized without this retry loop. If that is easy to implement it might allow a ton of cleanup in the testsuite.

Reply:
The answer is I completely flubbed it by not getting an the id passed in.  I have no idea how this worked when I ran it.

Yeah, I have a helper function in `job-list/job-list-helper.sh` but that is unique for the "large setup of jobs" that I do in `t2260-job-list.t` and `t2800-jobs-cmd.t`, so didn't quite work for this case of just a handful of jobs.

Perhaps I can stick this helper in `job-list/job-list-helper.sh` instead.

Reply:
I was more thinking of an RPC to job-list that responds when a job or list of jobs has reached a certain state. I haven't thought about how much trouble it would be to implement, but needing these for loops in tests is a little depressing and makes it seem like there's some interface missing. However, if it just more of a pain to add some interface to the module, then this is fine for now...

Reply:
@garlick are you thinking something sort of like `flux kvs wait`?  Insert a "version" or sorts into the stream?

Reply:
well not a version per se but a "sync cookie" I guess.  Since many operations in the job manager are synchronous with respect to appending events to the journal, one could 
1) do the thing
2) call job manager sync, get back identifier
3) call job-list sync with identifier, get back a response that indicates job-list is caught up to that point in the journal